internal/mode/static/nginx/config/generator.go

```diff
@@ -106,24 +114,39 @@ func generateCertBundleFileName(id dataplane.CertBundleID) string {
 	return filepath.Join(secretsFolder, string(id)+".crt")
 }
 
-func (g GeneratorImpl) generateHTTPConfig(conf dataplane.Configuration) file.File {
-	var c []byte
+func (g GeneratorImpl) generateHTTPConfig(conf dataplane.Configuration) []file.File {
+	files := make([]file.File, 0)
```

Comment:
we know the length of files after we constructed fileBytes map

so let's move the declaration down and use the known length.

```
files := make([]file.File, 0, len(fileBytes))
```